framework/utils/utils.py

```diff
@@ -1109,3 +1109,40 @@ def getAllSubclasses(cls):
     @ Out, getAllSubclasses, list of class objects for each subclass of cls.
   """
   return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in getAllSubclasses(s)]
+
+def which(cmd):
+  """
+    Emulate the which method in shutil
+    @ In, cmd, str, the exe to check
+    @ Out, which, str, the full path or None if not found
+  """
+  def _access_check(fn):
```

Comment:
should we add a description of this function?

Reply:
added